test/Microsoft.AspNetCore.Mvc.Core.Test/ModelBinding/Binders/ComplexTypeModelBinderTest.cs

```diff
@@ -350,6 +349,70 @@ public void CreateModel_InstantiatesInstanceOfMetadataType()
         }
 
         [Fact]
+        public void CreateModel_ForStructModelType_AsTopLevelObject_ThrowsException()
+        {
+            // Arrange
+            var bindingContext = new DefaultModelBindingContext
+            {
+                ModelMetadata = GetMetadataForType(typeof(PointStruct)),
+                IsTopLevelObject = true
+            };
+            var binder = CreateBinder(bindingContext.ModelMetadata);
+
+            // Act & Assert
+            var exception = Assert.Throws<InvalidOperationException>(() => binder.CreateModelPublic(bindingContext));
+            Assert.Equal(
+                string.Format(
+                    "Could not create an instance of type '{0}'. Model bound complex types must not be abstract or " +
+                    "value types and must have a parameterless constructor.",
+                    typeof(PointStruct).FullName),
+                exception.Message);
+        }
+
+        [Fact]
+        public void CreateModel_ForStructModelType_AsProperty_ThrowsException()
+        {
+            // Arrange
+            var bindingContext = new DefaultModelBindingContext
+            {
+                ModelMetadata = GetMetadataForProperty(typeof(Location), nameof(Location.Point)),
+                ModelName = nameof(Location.Point),
+                IsTopLevelObject = false
+            };
+            var binder = CreateBinder(bindingContext.ModelMetadata);
+
+            // Act & Assert
+            var exception = Assert.Throws<InvalidOperationException>(() => binder.CreateModelPublic(bindingContext));
+            Assert.Equal(
+                string.Format(
+                    "Could not create an instance of type '{0}'. Model bound complex types must not be abstract or " +
+                    "value types and must have a parameterless constructor. Alternatively, set the '{1}' property to" +
+                    " a non-null value in the '{2}' constructor.",
+                    typeof(PointStruct).FullName,
+                    nameof(Location.Point),
+                    typeof(Location).FullName),
+                exception.Message);
+        }
+
+        [Fact]
+        public void CreateModel_ByCustomModelBinderProvider_CanCreateModels_ForParameterlessConstructorTypes()
+        {
+            // Arrange
+            var bindingContext = new DefaultModelBindingContext
+            {
+                ModelMetadata = GetMetadataForType(typeof(AddressWithNoParameterlessConstructor))
+            };
+            var binder = new CustomComplexTypeModelBinder(new Dictionary<ModelMetadata, IModelBinder>());
+
+            // Act
+            var model = binder.CreateModelPublic(bindingContext);
```

Comment:
This test just calls the method you wrote in the test class 😲 . This really needs to call `BindModelAsync`.